corehq/apps/couch_sql_migration/casediff.py

```diff
@@ -732,12 +738,16 @@ def diff_case(sql_case, couch_case, dd_count):
     def diff(couch_json, sql_json):
         diffs = json_diff(couch_json, sql_json, track_list_indices=False)
         return filter_case_diffs(couch_json, sql_json, diffs, _diff_state)
+    case_id = couch_case['_id']
     sql_json = sql_case.to_json()
     dd_count("commcare.couchsqlmigration.case.diffed")
+    diffs = check_domains(case_id, couch_case, sql_json)
+    if diffs:
+        dd_count("commcare.couchsqlmigration.case.has_diff")
+        return couch_case, diffs
```

Comment:
Check domain just before the case will be diffed and possibly rebuilt. It turned out to require more code than I had hoped, although it allowed a few other nice cleanups now that the domain is on the statedb. This makes it much harder to imagine a scenario where the diff code would accidentally modify documents in the wrong domain.